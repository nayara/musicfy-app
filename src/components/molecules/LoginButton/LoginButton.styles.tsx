import { styled } from "styled-components";

const Label = styled.span`
  font-size: 2vh;
  color: ${({ theme }) => theme.colors.neutral.white};
  padding: 0.5rem 1rem;
  background-color: ${({ theme }) => theme.colors.highlight};
  border-radius: 0.5rem;
  margin: 20px;
  font-weight: bold;
  letter-spacing: 0.01rem;
`;

export { Label };
