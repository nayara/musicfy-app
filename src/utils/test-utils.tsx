import { RenderOptions, render } from "@testing-library/react";
import { FC, ReactElement, ReactNode } from "react";
import { Providers } from "src/providers";

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => {
  return <Providers>{children}</Providers>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };
