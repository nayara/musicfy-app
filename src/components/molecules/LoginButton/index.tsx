import Button from "../../../components/atoms/Button";
import * as Style from "./LoginButton.styles";

const handleLogin = () => {
  alert("to be implemented");
};

const LoginButton = () => {
  return (
    <Button onClick={handleLogin}>
      <Style.Label>Login</Style.Label>
    </Button>
  );
};

export default LoginButton;
