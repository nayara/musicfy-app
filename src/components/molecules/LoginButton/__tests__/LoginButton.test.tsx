import LoginButton from "..";
import { render } from "../.../../../../../utils/test-utils";

describe("<LoginButton />", () => {
  it("should render login button corretly", () => {
    const { container } = render(<LoginButton />);

    expect(container).toMatchSnapshot();
  });
});
