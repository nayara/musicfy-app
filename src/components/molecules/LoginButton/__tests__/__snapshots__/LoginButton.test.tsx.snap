// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginButton /> should render login button corretly 1`] = `
.c0 {
  background: none;
  color: inherit;
  border: none;
  padding: 0.5rem;
  font: inherit;
  cursor: pointer;
  outline: inherit;
}

.c1 {
  font-size: 2vh;
  color: #EEF1EF;
  padding: 0.5rem 1rem;
  background-color: #c1292e;
  border-radius: 0.5rem;
  margin: 20px;
  font-weight: bold;
  letter-spacing: 0.01rem;
}

<div>
  <button
    class="c0"
  >
    <span
      class="c1"
    >
      Login
    </span>
  </button>
</div>
`;
